node_management/manage_supra_nodes.sh

```diff
@@ -555,115 +665,72 @@ function start() {
-    fi
-
-    local bucket_name="mainnet"
-    # Define the custom endpoint for Cloudflare R2
-    local endpoint_url="https://4ecc77f16aaa2e53317a19267e3034a4.r2.cloudflarestorage.com"
+    install_rclone_if_missing
+    setup_rclone_config_if_missing
 
-    # Set AWS CLI credentials and bucket name based on the selected network
-    if [ "$NETWORK" == "mainnet" ]; then
-        export AWS_ACCESS_KEY_ID="ba3e917e8f1eb35772059f8eb3736cac"
-        export AWS_SECRET_ACCESS_KEY="432bbb569498de327299a40b6b2357dc282ff4158e939efbe6c75807c4885e3b"
-    elif [ "$NETWORK" == "testnet" ]; then
-        export AWS_ACCESS_KEY_ID="20f826bb30ea7205116e2adc7f19e34d"
-        export AWS_SECRET_ACCESS_KEY="1ad51161a013f1115381197929524cc1ff91fbccf016717652af8adf8fa2ed98"
-
-        if is_validator; then
-            bucket_name="testnet-validator-snapshot"
-        elif is_rpc; then
-            bucket_name="testnet-snapshot"
-        fi
+    if [[ "$1" == "--dry-run" ]]; then
+        echo "Running in dry-run mode..."
+        DRY_RUN="--dry-run"
+    else
+        DRY_RUN=""
     fi
 
+    # Determine bucket name
     if [ -n "$SNAPSHOT_SOURCE" ]; then
-        # The user overrode the default snapshot source via the optional parameter.
         bucket_name="$SNAPSHOT_SOURCE"
-    fi
-
-    local remote_root="s3://${bucket_name}/snapshots"
-    # The file that will be present in the `snapshots` directory of the bucket if the
-    # snapshot uploader is currently uploading an update.
-    local lock_file="$remote_root/UPLOAD_IN_PROGRESS"
-
-    # Sync the current state of the bucket.
-    sync_once "$endpoint_url" "$remote_root"
-
-    # Check if an upload is in progress. If it is, then the sync that we just completed
-    # will have terminated in an inconsistent state. Wait for the upload to finish, then
-    # sync one more time.
-    if aws s3 ls "$lock_file" --endpoint-url "$endpoint_url"; then
-        echo "A new snapshot is currently being uploaded to the bucket. "
-        echo -n "Waiting for the upload to complete. This may take 10-15 minutes..."
-
-        # Wait for the upload to complete.
-        while aws s3 ls "$lock_file" --endpoint-url "$endpoint_url"
-        do
+    elif [ "$NETWORK" == "mainnet" ]; then
+        bucket_name=$(is_validator && echo "mainnet-validator-snapshot" || echo "mainnet")
+    elif [ "$NETWORK" == "testnet" ]; then
+        bucket_name=$(is_validator && echo "testnet-validator-snapshot" || echo "testnet-snapshot")
+    fi
+
+    local remote_root="${RCLONE_REMOTE_NAME}:${bucket_name}/snapshots"
+    lock_dir="${remote_root}/"
+    lock_filename="UPLOAD_IN_PROGRESS"
+    sync_once "$remote_root"
+    echo "🔍 Checking for lock file: ${lock_dir}${lock_filename}"
+    if rclone lsf "$lock_dir" | grep -q "^${lock_filename}$"; then
+        echo "🚧 A new snapshot is currently being uploaded to the bucket."
+        echo -n "⏳ Waiting for the upload to complete. This may take 10–15 minutes..."
+
+        local retries=60
```

Comment:
The info message says that the upload can take up to 15 minutes, so we should wait longer than that before timing out.
```suggestion
        local retries=120
```